core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -336,6 +383,42 @@ namespace kagome::primitives {
     return decodeVariant<Invalid, Valid, Unknown>(stream);
   }
 
+  outcome::result<Buffer> ScaleCodecImpl::encodeInherentData(
+      const InherentData &inherentData) const {
+    auto &data = inherentData.getDataCollection();
```

Comment:
```suggestion
    const auto &data = inherentData.getDataCollection();
```
If I am not wrong, you do not mutate `data` here, so it can (and should) be constref

Reply:
I believe it's already a cons ref, as inherentData is a const ref and it would be impossible here to obtain a mutable reference on its field.